0xProject/0x-monorepo
# OrderWatcher Fixes

## Description

This PR:
- Makes OrderWatcher resilient against block re-orgs, and other blockchain quirks (e.g if transactions skip the nodes mempool and go straight to being confirmed). All these scenarios now lead to proper event emissions.
- Does not close the OrderWatcher subscription if it's underlying Blockstream instance encounters inconsistent responses from the backing Ethereum node.

## Testing instructions

- Run the tests.
- Start running an OrderWatcher, fetch all the orders on ERCDEX, add them to the OrderWatcher, visually verify that the OrderWatcher emits events, does not go down if blockstream errors are encountered.



## Types of changes




## Checklist:
